Formula/singular.rb

```diff
@@ -0,0 +1,34 @@
+class Singular < Formula
+  desc "Computer algebra system for polynomial computations"
+  homepage "https://www.singular.uni-kl.de/"
+  url "http://www.mathematik.uni-kl.de/ftp/pub/Math/Singular/src/4-1-0/singular-4.1.0p3.tar.gz"
+  version "4.1.0p3"
+  sha256 "440164c850d5a1575fcbfe95ab884088d03c0449570d40f465611932ffd0bf80"
+  head "https://github.com/Singular/Sources.git"
+
+  depends_on "autoconf" => :build
+  depends_on "automake" => :build
+  depends_on "libtool" => :build
+  depends_on "gmp"
+  depends_on "ntl"
+
+  def install
+    system "./autogen.sh"
+    system "./configure", "--disable-debug",
+                          "--disable-dependency-tracking",
+                          "--disable-silent-rules",
+                          "--prefix=#{prefix}"
+    system "make", "install"
+  end
+
+  test do
+    testinput = <<~EOS
+      ring r = 0,(x,y,z),dp;
+      poly p = x;
+      poly q = y;
+      poly qq = z;
+      p*q*qq;
+    EOS
+    assert_match "xyz", pipe_output(bin/"Singular", testinput, 0)
```

Comment:
prefer `"#{bin}/Singular"` to `bin/"Singular"` for the first argument to pipe_output and shell_output